src/hotspot/share/gc/shenandoah/shenandoahCollectorPolicy.hpp

```diff
@@ -40,7 +40,8 @@ class ShenandoahCollectorPolicy : public CHeapObj<mtGC> {
 private:
   size_t _success_concurrent_gcs;
   size_t _success_degenerated_gcs;
-  size_t _success_full_gcs;
+  // Written by control thread, read by mutators
+  volatile size_t _success_full_gcs;
   size_t _alloc_failure_degenerated;
   size_t _alloc_failure_degenerated_upgrade_to_full;
```

Comment:
In GenShen tip, `_alloc_failure_degenerated_upgrade_to_full` is also declared volatile.

Reply:
`_alloc_failure_degenerated_upgrade_to_full` is only changed on a safepoint, so it shouldn't need any additional synchronization.

Reply:
OK, as long as this resolves in genshen tip.